lib/foreman_tasks_core/otp_manager.rb

```diff
@@ -17,10 +17,12 @@ def passwords
         @password ||= {}
       end
 
-      def authenticate(hash)
+      def authenticate(hash, clear: true)
         plain = Base64.decode64(hash)
         username, otp = plain.split(':', 2)
-        drop_otp(username, otp)
+        password_matches = passwords[username] == otp
+        drop_otp(username, otp) if clear
+        password_matches
```

Comment:
How about something like
```ruby
password_matches = passwords[username] == otp
passwords.delete(username) if clear && password_matches
password_matches
```
to avoid comparing the otp twice (and an unnecessary function call)